tests/mypy_test.py

```diff
@@ -151,25 +151,15 @@ def parse_version(v_str: str) -> tuple[int, int]:
     return int(m.group(1)), int(m.group(2))
 
 
-def add_files(files: list[Path], seen: set[str], root: StrPath, name: str, args: TestConfig) -> None:
+def add_files(files: list[Path], seen: set[str], module: Path, args: TestConfig) -> None:
     """Add all files in package or module represented by 'name' located in 'root'."""
-    full = Path(root, name)
-    mod, ext = full.stem, full.suffix
-    if ext in [".pyi", ".py"]:
-        if match(full, args):
-            seen.add(mod)
-            files.append(full)
-    elif (full / "__init__.pyi").is_file() or (full / "__init__.py").is_file():
-        for r, ds, fs in os.walk(full):
-            ds.sort()
-            fs.sort()
-            for f in fs:
-                m, x = os.path.splitext(f)
-                if x in [".pyi", ".py"]:
-                    fn = Path(r, f)
-                    if match(fn, args):
-                        seen.add(mod)
-                        files.append(fn)
+    if module.is_file() and module.suffix == ".pyi":
+        files.append(module)
+        seen.add(module.stem)
+    else:
+        to_add = sorted(module.rglob("*.pyi"))
```

Comment:
I believe you need `**/*.pyi`; this will not recurse into subdirectories.

Reply:
I think it will, because we're using `rglob`, not `glob` :) https://docs.python.org/3/library/pathlib.html#pathlib.Path.rglob

I checked locally, and it collects the same number of files (for both the stdlib and the third-party packages) as we currently do in CI

Reply:
I did some more testing -- I'm pretty confident that the code proposed in this PR collects exactly the same list of files as the code on typeshed master. The order of the `files` list appears to be slightly different, but personally I don't think that matters much?

Reply:
Thanks for confirming!